core-api/src/main/java/org/silverpeas/core/variables/VariablePeriod.java

```diff
@@ -0,0 +1,97 @@
+ * License, or (at your option) any later version.
+ *
+ * As a special exception to the terms and conditions of version 3.0 of
+ * the GPL, you may redistribute this Program in connection with Free/Libre
+ * Open Source Software ("FLOSS") applications as described in Silverpeas's
+ * FLOSS exception.  You should have received a copy of the text describing
+ * the FLOSS exception, and it is also available here:
+ * "https://www.silverpeas.org/legal/floss_exception.html"
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.silverpeas.core.variables;
+
+import org.silverpeas.core.date.period.Period;
+import org.silverpeas.core.persistence.datasource.model.identifier.UuidIdentifier;
+import org.silverpeas.core.persistence.datasource.model.jpa.SilverpeasJpaEntity;
+import org.silverpeas.core.security.Securable;
+import org.silverpeas.core.util.WebEncodeHelper;
+
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.FetchType;
+import javax.persistence.JoinColumn;
+import javax.persistence.ManyToOne;
+import javax.persistence.NamedQueries;
+import javax.persistence.NamedQuery;
+import javax.persistence.Table;
+import java.util.Date;
+
+@Entity
+@Table(name = "sb_variables_period")
+@NamedQueries({
+    @NamedQuery(name = "currentPeriods", query = "select p from VariablePeriod p where p" +
+        ".startDate <= :today and :today <= p.endDate order by p.variable.label ASC")})
+public class VariablePeriod extends SilverpeasJpaEntity<VariablePeriod, UuidIdentifier>
+    implements Securable {
+
+  @Column(nullable = false)
+  private String value;
+
+  @Column
+  private Date startDate;
+
+  @Column
+  private Date endDate;
+
+  @ManyToOne(fetch = FetchType.EAGER, optional = false)
+  @JoinColumn(name = "variableId", referencedColumnName = "id", nullable = false)
+  private Variable variable;
+
+  protected VariablePeriod() {
+    // default constructor for the persistence engine
+  }
+
+  public VariablePeriod(String value, Period period) {
+    this.value = value;
+    this.startDate = period.getBeginDate();
+    this.endDate = period.getEndDate();
+  }
+
+  public Period getPeriod() {
+    return Period.from(startDate, endDate);
+  }
+
+  public String getValue() {
+    return value;
+  }
+
+  public void setVariable(Variable value) {
+    variable = value;
+  }
+
+  public Variable getVariable() {
+    return variable;
+  }
+
+  public String getValueForHTML() {
+    return WebEncodeHelper.javaStringToHtmlParagraphe(getValue());
+  }
+
+  public VariablePeriod setId(String id) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Add the "@Override" annotation above this method signature [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS1161)